packages/desktop/components/popups/WithdrawFromL2Popup.svelte

```diff
@@ -0,0 +1,197 @@
+        coinType: Number(getCoinType()),
+        account: getSelectedAccount().index,
+        change: 0,
+        addressIndex: 0,
+    }
+
+    let error = ''
+    let address: string | undefined = undefined
+    let isWithdrawing = false
+    const { isStrongholdLocked } = $activeProfile
+
+    $: withdrawOnLoad && address && !$isStrongholdLocked && withdrawFromL2()
+
+    function onCancelClick(): void {
+        closePopup()
+    }
+
+    async function onWithdrawFromL2Click(): Promise<void> {
+        if ($isSoftwareProfile && $isStrongholdLocked) {
+            openUnlockStrongholdPopup()
+        } else {
+            await handleAction(withdrawFromL2)
+        }
+    }
+
+    async function handleAction(callback: () => Promise<void>): Promise<void> {
+        try {
+            error = ''
+
+            if ($isActiveLedgerProfile && !$ledgerNanoStatus.connected) {
+                displayNotificationForLedgerProfile('warning')
+                return
+            }
+
+            await callback()
+        } catch (err) {
+            error = localize(err.error)
+
+            if ($isActiveLedgerProfile) {
+                displayNotificationForLedgerProfile('error', true, true, err)
+            } else {
+                showAppNotification({
+                    type: 'error',
+                    message: localize(err.error),
+                })
+            }
+        }
+    }
+
+    async function withdrawFromL2(): Promise<void> {
+        isWithdrawing = true
+        if ($isActiveLedgerProfile && !$ledgerNanoStatus.connected) {
+            isWithdrawing = false
+            displayNotificationForLedgerProfile('warning')
+            return
+        }
+
+        const nonce = await getNonceForWithdrawRequest(address)
+        if (!nonce) {
+            isWithdrawing = false
+            displayNotificationForLedgerProfile('warning')
+            return
+        }
+        let withdrawRequest = await getLayer2WithdrawRequest(withdrawableAmount.toString(), nonce, bip44Chain)
+        // get gas estimate for request with hardcoded amounts
+        const gasEstimatePayload = await getEstimatedGasForOffLedgerRequest(withdrawRequest.request)
+        const minRequiredStorageDeposit: number = await getRequiredStorageDepositForMinimalBasicOutput()
+        //  The "+1" is due to an optimization in WASP nodes.
+        const gasEstimate = gasEstimatePayload.gasFeeCharged + 1
+        if (withdrawableAmount > Number(minRequiredStorageDeposit) + Number(gasEstimate)) {
+            // Create new withdraw request with correct gas budget
+            withdrawRequest = await getLayer2WithdrawRequest(
+                (withdrawableAmount - gasEstimate).toString(),
+                nonce,
+                bip44Chain,
+                gasEstimate.toString()
+            )
+        } else {
+            isWithdrawing = false
+            showErrorNotification(localize('error.send.notEnoughBalance'))
+            return
+        }
+
+        try {
```

Comment:
In function `getLayer2WithdrawRequest`, we need to accept de signature with the ledger device but the verification popup does not open

1. This `try catch` should be placed just above the first call to the `getLayer2WithdrawRequest` function so that if the tx is canceled the `withdraw` button does not remain in an infinite loop
2. I don't know if we can open the verification popup directly, but if not we could put a notification like this at least 
`if ($isActiveLedgerProfile && $ledgerNanoStatus.connected) {
                    displayNotificationForLedgerProfile('info', null, null, 'Look your ledger device to sign the transaction')
                }`
wdyt?